sonar-python-plugin/src/test/java/org/sonar/plugins/python/pylint/PylintSensorTest.java

```diff
@@ -0,0 +1,238 @@
+    assertThat(first.severity()).isEqualTo(Severity.MAJOR);
+    IssueLocation firstPrimaryLoc = first.primaryLocation();
+    assertThat(firstPrimaryLoc.inputComponent().key()).isEqualTo(PYLINT_FILE);
+    assertThat(firstPrimaryLoc.message())
+      .isEqualTo("message");
+
+    externalIssues = executeSensorImporting(7, 9, "pylint_names_in_brackets.txt");
+    assertThat(externalIssues).hasSize(1);
+    first = externalIssues.get(0);
+    assertThat(first.ruleKey()).hasToString("external_pylint:C0111");
+    assertThat(first.type()).isEqualTo(RuleType.CODE_SMELL);
+    assertThat(first.severity()).isEqualTo(Severity.MAJOR);
+    firstPrimaryLoc = first.primaryLocation();
+    assertThat(firstPrimaryLoc.inputComponent().key()).isEqualTo(PYLINT_FILE);
+    assertThat(firstPrimaryLoc.message())
+      .isEqualTo("Missing docstring");
+  }
+
+  @Test
+  public void no_issues_unknown_files() throws IOException {
+    List<ExternalIssue> externalIssues = executeSensorImporting(7, 9, "pylint_report_unknown_files.txt");
+    assertThat(externalIssues).isEmpty();
+  }
+
+  @Test
+  public void no_issues_with_invalid_report_path() throws IOException {
+    List<ExternalIssue> externalIssues = executeSensorImporting(7, 9, "invalid-path.txt");
+    assertThat(externalIssues).isEmpty();
+    assertThat(onlyOneLogElement(logTester.logs(LoggerLevel.ERROR)))
+      .startsWith("No issues information will be saved as the report file '")
+      .contains("invalid-path.txt' can't be read.");
+  }
+
+  private static List<ExternalIssue> executeSensorImporting(int majorVersion, int minorVersion, @Nullable String fileName) throws IOException {
+    Path baseDir = PROJECT_DIR.getParent();
+    SensorContextTester context = SensorContextTester.create(baseDir);
+    try (Stream<Path> fileStream = Files.list(PROJECT_DIR)) {
+      fileStream.forEach(file -> addFileToContext(context, baseDir, file));
+      context.setRuntime(SonarRuntimeImpl.forSonarQube(Version.create(majorVersion, minorVersion), SonarQubeSide.SERVER, SonarEdition.DEVELOPER));
+      if (fileName != null) {
+        String path = PROJECT_DIR.resolve(fileName).toAbsolutePath().toString();
+        context.settings().setProperty("sonar.python.pylint.reportPaths", path);
+      }
+      pylintSensor.execute(context);
+      return new ArrayList<>(context.allExternalIssues());
+    }
+  }
+
+  private static void addFileToContext(SensorContextTester context, Path projectDir, Path file) {
+    try {
+      String projectId = projectDir.getFileName().toString() + "-project";
+      context.fileSystem().add(TestInputFileBuilder.create(projectId, projectDir.toFile(), file.toFile())
+        .setCharset(UTF_8)
+        .setLanguage(language(file))
+        .setContents(new String(Files.readAllBytes(file), UTF_8))
+        .setType(InputFile.Type.MAIN)
+        .build());
+    } catch (IOException e) {
+      throw new IllegalStateException(e);
+    }
```

Comment:
Not sure I understand why we are doing this, `executeSensorImporting` has already declared that it may throw `IOException`

Reply:
After double checking it seems that it's because this method is called in a `forEach` statement, and because of that the `IOException` it might throw would not be handled by the `executeSensorImporting` throw declaration.